src/components/app/data/hooks/useBFF.js

```diff
@@ -1,36 +1,45 @@
 import { useLocation, useParams } from 'react-router-dom';
 import { useQuery } from '@tanstack/react-query';
-import { resolveBFFQuery } from '../../routes/data/utils';
+import { logError } from '@edx/frontend-platform/logging';
+import useEnterpriseCustomer from './useEnterpriseCustomer';
+import { resolveBFFQuery } from '../queries';
 
 /**
  * Uses the route to determine which API call to make for the BFF
  * Populates the queryKey with the appropriate enterprise customer uuid once BFF call is resolved
- * @param queryOptions
+ * @param bffQueryAdditionalParams - additional fields to pass into a matched BFF query call
+ * @param bffQueryOptions - the queryOptions specifically for the matched BFF query call
+ * @param fallbackQueryConfig - if a route is not compatible with the BFF layer, this field
+ * allows you to pass a fallback query endpoint to call in lieu of an unmatched BFF query
  * @returns  {Types.UseQueryResult}} The query results for the routes BFF.
  */
-export function useBFF(queryOptions = {}) {
-  const { select, ...queryOptionsRest } = queryOptions;
+export default function useBFF(
+  bffQueryAdditionalParams = {},
+  bffQueryOptions = {},
+  fallbackQueryConfig = null,
+) {
```

Comment:
nit: so it's easier to pass the optional args without worrying about argument order, and so it's possible to parse which argument is which at a glance.

```suggestion
export default function useBFF({
  bffQueryAdditionalParams = {},
  bffQueryOptions = {},
  fallbackQueryConfig = null,
}) {
```

Before:

```jsx
return useBFF(
  {},
  {
    select: (data) => transformSubscriptionsData(data?.enterpriseCustomerUserSubsidies?.subscriptions),
  },
  {
    ...querySubscriptions(enterpriseCustomer.uuid),
    ...queryOptions,
  },
);
```

After:

```jsx
return useBFF({
  bffQueryOptions: {
    select: (data) => transformSubscriptionsData(data?.enterpriseCustomerUserSubsidies?.subscriptions),
  },
  fallbackQueryConfig: {
    ...querySubscriptions(enterpriseCustomer.uuid),
    ...queryOptions,
  },
})
```